src/main/java/net/sf/jabref/gui/entryeditor/EntryEditor.java

```diff
@@ -467,8 +474,29 @@ public void mouseClicked(MouseEvent e) {
                 contentSelectors.add(ws);
                 controls.add(ws, BorderLayout.NORTH);
             }
-            controls.add(JournalAbbreviationsUtil.getNameSwitcher(this, editor, panel.undoManager),
-                    BorderLayout.SOUTH);
+
+            // Button to toggle abbreviated/full journal names
+            JButton button = new JButton(Localization.lang("Toggle abbreviation"));
+            button.setToolTipText(ABBREVIATION_TOOLTIP_TEXT);
+            button.addActionListener(new ActionListener() {
+
+                @Override
+                public void actionPerformed(ActionEvent actionEvent) {
+                    String text = editor.getText();
+                    if (Abbreviations.journalAbbrev.isKnownName(text)) {
+                        String s = Abbreviations.journalAbbrev.getNextAbbreviation(text).orElse(text);
```

Comment:
Can you move this line so that instead of `Abbreviations.journalAbbrev.getNextAbbreviation(text).orElse(text);` one can call `Abbreviations.journalAbbrev.toggleAbbreviation(text);`?

The aim is to have as much code as possible in logic and independent of the GUI classes.